src/BoundsInference.cpp

```diff
@@ -369,9 +369,9 @@ class BoundsInference : public IRMutator {
             // Check if it succeeded
             string result_name = unique_name('t');
             Expr result = Variable::make(Int(32), result_name);
-            vector<Expr> error_message = vec<Expr>("Bounds inference call to external func " + extern_name +
-                                                   " returned non-zero value:", result);
-            Stmt check = AssertStmt::make(EQ::make(result, 0), error_message);
+            Expr error = Call::make(Int(32), "halide_error_bounds_inference_call_failed",
```

Comment:
If they are used in precisely two places, then no, it probably isn't worth hoisting.